inst/htmlwidgets/plotly.js

```diff
@@ -263,10 +263,13 @@ HTMLWidgets.widget({
         for (var i = 0; i < attrsToAttach.length; i++) {
           var attr = trace[attrsToAttach[i]];
           if (Array.isArray(attr)) {
-              // pointNumber can be an array (e.g., heatmaps)
-              // TODO: can pointNumber be 3D?
-              obj[attrsToAttach[i]] = typeof pt.pointNumber === "number" ? 
-                attr[pt.pointNumber] : attr[pt.pointNumber[0]][pt.pointNumber[1]];
+            var ptNums = pt.pointNumber || pt.pointNumbers;
+            if (typeof ptNums === "number") {
+              ptNums = [ptNums];
```

Comment:
In this case, it will make an array of a single number.

The final store value produced in line 271 will be an array of `attr` values.  The final store value above in _old_ 269 is only the `attr` value if `pt.pointNumber` is a number.

Is this allowable?

Reply:
Ahh, yes, good catch. It's not allowable in the sense that it changes the event data emitted user. Do you like this pattern better than "unboxing" arrays of length 1 after-the-fact?

```js
var ptNums = pt.pointNumber || pt.pointNumbers;
if (typeof ptNums === "number") {
    obj[attrsToAttach[i]] = ptNums;
} else if (Array.isArray(ptNums)) {
     obj[attrsToAttach[i]] = ptNums.map(function(i) { return attr[i]; });
}
```

Reply:
Not a fan of unboxing after the fact.  I'd even take it a step further and only use the attributes where possible and not overwrite `i` in the map (not necessary, but nice).

```js
if (typeof ptNums === "number") {
    obj[attrsToAttach[i]] = attr[pt.pointNumber];
} else if (Array.isArray(ptNums)) {
    obj[attrsToAttach[i]] = pt.pointNumbers.map(function(ptNum) { return attr[ptNum]; });
}
```

or 

```js
obj[attrsToAttach[i]] = (typeof ptNums === "number") ? 
  attr[pt.pointNumber] :
  pt.pointNumbers.map(function(ptNum) { return attr[ptNum]; });
```